app/src/main/res/layout/activity_microscope_quality.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".ui.microscope_quality.MicroscopeQualityActivity">
+
+    <include
+        layout="@layout/toolbar"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Why is a ConstraintLayout needed in another ConstraintLayout? 🙈

Reply:
XD Just a matter of organisation, with some subtle consequences, like the inner ConstraintLayout padding, which replaced the padding that would be added to every element within it... Is that as bad as an anti-pattern? How could we tackle that attribute's repetition, if not with this approach? Thanks!

Reply:
I see. I think that there are two possible solutions: 
- One outer ConstraintLayout with Guidelines for the padding.
or
- Use a LinearLayout with vertical orientation instead of the inner ConstraintLayout.

Probably the latter is easier/faster to achieve.

Reply:
The first one is not really an option, as the padding would be applied to all elements, including the toolbar and the action bar (I tried that when going through this XD).
I had thought about the second one, but what is the advantage on that? (I'm asking, not challenging) I'd expect the LinearLayout to be better performant than the ConstraintLayout, but what I read is that ConstraintLayout should be used whenever possible (could not get to a conclusion on the reason - maybe to try to get rid of nested layouts like this one?). Thanks!!

Reply:
Is there any way to make the LinearLayout fill the whole screen vertically without needing to use a spacer? (ideally the inner ConstraintLayout should fill its parent)

Reply:
> The first one is not really an option, as the padding would be applied to all elements

With a [`Guideline`](https://constraintlayout.com/basics/guidelines.html), you can choose what elements "have" padding, and which not. 

> ConstraintLayout should be used whenever possible

The idea of the `ConstraintLayout` is to have a flat hierarchy, so yes, it should be used whenever possible, but having nested ones defeats its purpose. And yes, the ultimate price that you are paying is performance.

> Is there any way to make the LinearLayout fill the whole screen vertically without needing to use a spacer? (ideally the inner ConstraintLayout should fill its parent)

Yes, let the `LinearLayout` have `android:layout_height="match_parent"`, and then on the widget that you want to take up all the available space put the following:
```xml
        android:layout_height="0dp"
        android:layout_weight="1"
```

Reply:
Great tips there!! Thanks!